core/modules/nfv_ctrl.cc

```diff
@@ -111,6 +111,16 @@ void NFVCtrl::ReleaseSwQ(int q_id) {
   bess::ctrl::sw_q_state[q_id]->up_core_id = DEFAULT_INVALID_CORE_ID;
 }
 
+void DumpQueueBatch(struct llring* q) {
+  bess::PacketBatch *batch = reinterpret_cast<bess::PacketBatch *>
```

Comment:
This will lead to the memory leak problem because |batch| is not released after allocation.
I'd suggest not creating a new packet batch.
Instead, just keep add packets to the |batch| that comes with RunTask(ctx, batch, arg) untile 1) |batch| is full; 2) all dumping software queues are empty.